yarn-project/end-to-end/src/spartan/utils.ts

```diff
@@ -350,7 +350,11 @@ export async function awaitL2BlockNumber(
     await sleep(1000);
     tips = await rollupCheatCodes.getTips();
   }
-  logger.info(`Reached L2 Block ${tips.pending}`);
+  if (Date.now() >= endTime) {
```

Comment:
```suggestion
  if (tips.pending < blockNumber) {
```
So we don't fail if we got in there in the very last second!